//Grupo 15: Pablo Arlandis Ocaña 
package tp1.control.commands;

import tp1.exceptions.CommandExecuteException;
import tp1.logic.*;
import tp1.view.GameView;
import tp1.view.Messages;

public class ExitCommand extends NoParamsCommand{
	
	private static final String NAME = Messages.COMMAND_EXIT_NAME;
	private static final String SHORTCUT = Messages.COMMAND_EXIT_SHORTCUT;
	private static final String DETAILS = Messages.COMMAND_EXIT_DETAILS;
	private static final String HELP = Messages.COMMAND_EXIT_HELP;

	ExitCommand() {
		super(NAME, SHORTCUT, DETAILS, HELP); 
	}

	@Override
	public void execute(GameModel game, GameView view) throws CommandExecuteException{
		// You should let the game know that you are leaving so that 
		// if it needs to close something, it can close it and finish.
	    game.exit(); 	
	}

}
